replica/table.cc

```diff
@@ -1564,8 +1564,6 @@ future<bool> table::perform_offstrategy_compaction(std::optional<tasks::task_inf
 }
 
 future<> table::perform_cleanup_compaction(compaction::owned_ranges_ptr sorted_owned_ranges, std::optional<tasks::task_info> info) {
-    co_await flush();
-
```

Comment:
If we push the flush_mode flag down here, we can flush conditionally e.g. `if (_flush_mode != flush_mode::skip)`

Reply:
see https://github.com/scylladb/scylla-ccm/pull/541 for an example where we have too many options.

Reply:
Having this flag follow the logic of major compaction and prserves the preexisting default semantics of flush prior to cleanup, so I don't think it generates much complexity.

Reply:
`perform_cleanup_compaction()` is more like an internal function. apart from the tests, it is only used by `table_cleanup_keyspace_compaction_task_impl::run()`. so i don't quite it's important to preserve its existing behavior.

if you insist on passing `flush_mode` all the way from we have to update following classes

- `table_cleanup_keyspace_compaction_task_impl`
- `shard_cleanup_keyspace_compaction_task_impl`

do you still think it is a good bargain ?

Reply:
instead of promoting `flush_mode` to a place which can be shared by both `task_manager_module.hh` and `database.hh`, i switched to `table::do_flush`. for two reasons:

1. it is more consistent with existing `table::compact_all_sstables()`. this is very important.
2. we don't have to find a new home for `flush_mode`.